spec/unit/knife/cookbook_metadata_spec.rb

```diff
@@ -174,6 +175,15 @@
         expect(@stderr.string).to match /#{description} blah/im
       end
     end
+    it "should raise a syntax error on invalid JSON" do
+      expect(File).to receive(:exist?).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(true)
+      expect(IO).to receive(:read).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(@invalid_json_data)
```

Comment:
A link or two showing what you consider to be better implementations would be really helpful.